File: search/bubble_sort/python/bubble_sort.py
def bubble_sort(data):
    """
    This method sorts a given list of numbers in ascending order, using the
    bubble sort algorithm.
    :param data: The list of numbers to sort.
    :return: The given list, sorted in ascending order.
    """
    
    n = len(data)
    swapped = True

    while swapped: # While "work is done" inside the loop;
        swapped = False

        for i in range(1, n):
            if data[i - 1] > data[i]: # If the elements are out of order, swap them.
                tmp = data[i - 1]
                data[i - 1] = data[i]
                data[i] = tmp
                swapped = True # Show something happened in the loop.

    return data
        

File: data_structures/Arrays/JS/bubbleSort.js
//Bubble sort
const bubbleSort = (arr) => {
	//Length of the array
	let n = arr.length;
  
	for (let i = 0; i < n - 1; i++) {
	  for (let j = 0; j <= n - i - 1; j++) {
		//Swap the numbers
		if (arr[j] > arr[j + 1]) {
		  [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
		}
	  }
	}
  
	return arr;
  };
  
  // Usage
  console.log(bubbleSort([10, 9, -1, -2, 0, 22]));